src/operators/courseItemOperations.ts

```diff
@@ -0,0 +1,61 @@
+import CourseItem, { CourseItemInterface } from '../models/courseItem';
+import { v4 as uuidv4 } from 'uuid';
+
+export const create = async (courseItem: CourseItemInterface) => {
+	try {
+		const hashKey = uuidv4();
+		courseItem.id = hashKey;
+		console.log('id: ' + courseItem.id);
+		await CourseItem.create(courseItem);
+	} catch (err) {
+		console.error(err);
+	}
+};
+
+// NOTE: any reference to key is referring to the hash key, or the `id` parameter in all the models.
+export const read = async (key: string, user: string) => {
+	try {
+		const courseItem = await CourseItem.get(key);
+		if (courseItem && courseItem.owner == user) {
+			return courseItem;
+		} else {
+			throw 'ERROR - could not read courseItem with key ' + key;
+		}
+	} catch (err) {
+		console.error(err);
+	}
+};
+
+export const update = async (
+	data: Partial<CourseItemInterface>,
+	userID: string
+) => {
+	try {
+		if (data.id) {
+			const key = data.id;
+			if (await checkUserID(key, userID)) {
+				await CourseItem.update(data);
+			} else {
+				throw "ERROR - userID doesn't match";
+			}
+		}
```

Comment:
I think we should do a `get` operation first and then an update to make sure the item already exists. (unless there is a way to block creating an object when we call update for something that doesn't exist)

Reply:
Since it calls `checkUserID` first I believe if the item doesn't exist, it won't continue with the update since it isn't able to check if `document.owner === userID`. Is this what you mean?